checkov/runner_filter.py

```diff
@@ -185,6 +186,38 @@ def should_run_check(
         logging.debug(f'Should run check {check_id}: {result}')
         return result
 
+    def should_run_check_for_file(self, check_id: str, file_full_path: str, root_folder: str | None) -> bool:
+        """
+        Check if skip check_id for a certain file_types, according to given path pattern
+        """
+        # Check that we have skip check with a path pattern
+        if not self.skip_checks:
+            return True
+
+        skip_check_with_rule = next(iter([skip_check for skip_check in self.skip_checks if check_id in skip_check]), None)
+        if not skip_check_with_rule:
+            return True
+
+        # skip_check_with_rule pattern should be as following (All upper case params is configured by user:
+        # "{CHECK_ID}:{DIRECTORY_RELATIVE_TO_ROOT}/{FILE_NAME_PATTERN}.{FILE_TYPE}
+        splitted_check = skip_check_with_rule.split(":")
+        # In case it's not expected pattern
+        if len(splitted_check) != 2:
+            return True
+        # Creating a Regex pattern according to User Input
+        pattern = splitted_check[1]
+        full_regex_pattern = fr"^{root_folder}/{pattern}" if root_folder else pattern
```

Comment:
@pazbechor 
1. This can be supplied multiple times, so let's not take only the 1st via `next`
2. Let's compute this once per run, in the constructor, and have a list (empty or non-empty) of regexes to filter

Reply:
Done, pls take a look :)